src/main/java/com/powsybl/openloadflow/network/LfBus.java

```diff
@@ -99,6 +104,10 @@ public interface LfBus extends LfElement {
 
     double getMaxQ();
 
+    Optional<QLimitType> getQLimitType();
+
+    void setQLimitType(QLimitType qLimitType);
```

Comment:
not sure if issue or not

we never do setQLimitType(null) when switching PQ->PV.

I do not think it would be required, Just we need to be aware e.g. during debug session that a bus may be PV now but still with an old QLimitType that was set from a previous outerloop.

Reply:
Yes that would be more safe to set to null when switching PV